code/__HELPERS/icons.dm

```diff
@@ -1459,3 +1459,85 @@ GLOBAL_LIST_EMPTY(friendly_animal_types)
 	animate(src, pixel_x = pixel_x + shiftx, pixel_y = pixel_y + shifty, time = 0.2, loop = duration)
 	pixel_x = initialpixelx
 	pixel_y = initialpixely
+
+/**
+ * Returns a list that contains the leftmost, rightmost, bottom and top pixels in an atom's icon
+ */
+/atom/proc/get_bounding_box()
+	var/list/return_list = list()
+	for(var/found_type in GLOB.cached_image_borders)
+		if(found_type == "[type]_[icon_state]")
+			return_list["left"] = GLOB.cached_image_borders[found_type]["left"]
+			return_list["right"] = GLOB.cached_image_borders[found_type]["right"]
+			return_list["top"] = GLOB.cached_image_borders[found_type]["top"]
+			return_list["bottom"] = GLOB.cached_image_borders[found_type]["bottom"]
+			return return_list
+
+	var/icon/tempicon = icon(icon, icon_state, dir, 1)
+	tempicon = getFlatIcon(src)
+	var/x_coord = 1
+	var/y_coord = 1
+	var/height = tempicon.Height()
+	var/width = tempicon.Width()
+	var/left_border = width
+	var/right_border = 1
+
+	tempicon.Scale(width, 1) //Flatten the icon into a 1 pixel high line
+	height = tempicon.Height()
+	while(y_coord <= height)
```

Comment:
Does it actually flatten the icon? I might be missunderstanding the proc, but reading the documentation I'm skeptical